### PR TITLE
Fix "time_to_second" helper

### PR DESCRIPTION
**Pull request by Jedoku**

### CODE REVIEW

File: lib/auxiliary/time_helper.rb [hunk 1]
```diff
@@ -4,9 +4,11 @@
 module AutoHCK
```
Comment (kostyanf14) [inline lib/auxiliary/time_helper.rb:4]: `Signed-off-by: Vitalii Chulak <vchulakv@gmail.com>`: use @daynix mail there

`fix time to second helper`:  commit title starts with an uppercase letter
`time_to_second` is a function, so better to write its name as is in the commit message without any splitting

**Review by kostyanf14 — CHANGES_REQUESTED** [review]

File: lib/auxiliary/time_helper.rb [hunk 2]
```diff
@@ -4,9 +4,11 @@
...
   # Helper module
   module Helper
-    def time_to_seconds(time)
-      time.split(':').reverse.map.with_index { |a, i| a.to_i * (60**i) }
-          .reduce(:+)
+    def time_to_seconds(time_str)
+      days, time_part = time_str.split('.')
```
Comment (akihikodaki) [inline lib/auxiliary/time_helper.rb:8]:
```suggestion
      days, time_part = time_str.split('.', 2)
```

Looking at .NET's own parser:
https://github.com/dotnet/runtime/blob/v8.0.1/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.TimeSpan.C.cs

There are the following possible formats:
- dd
- hh:mm
- dd.hh:mm
- hh:mm:ss
- dd.hh:mm:ss
- hh:mm:ss.fffffff
- dd.hh:mm:ss.fffffff

As shown here, `.` may appear before the second fraction too. We don't want to split it.

Also let's name `days` as `days_str` to make it consistent with `time_str` and to make it clear it is a string unlike `hours`, `minutes` and `seconds`.

File: lib/auxiliary/time_helper.rb [hunk 3]
```diff
@@ -4,9 +4,11 @@
...
+      hours, minutes, seconds = time_part.split(':').map(&:to_i)
+      total_hours = (days.to_i * 24) + hours
+      (total_hours * 3600) + (minutes * 60) + seconds
```
Comment (akihikodaki) [inline lib/auxiliary/time_helper.rb:11]: Why don't you use the original code for `time_part` parsing?

File: lib/auxiliary/time_helper.rb [hunk 4]
```diff
@@ -4,9 +4,11 @@
 module AutoHCK
   # Helper module
   module Helper
-    def time_to_seconds(time)
-      time.split(':').reverse.map.with_index { |a, i| a.to_i * (60**i) }
-          .reduce(:+)
+    def time_to_seconds(time_str)
+      days, time_part = time_str.split('.')
+      hours, minutes, seconds = time_part.split(':').map(&:to_i)
```
Comment (akihikodaki) [inline lib/auxiliary/time_helper.rb:9]: Use `to_f` to parse the second fraction.

**Review by akihikodaki — CHANGES_REQUESTED** [review]
Please add tests if you are not in a hurry. Use the test cases from .NET:
https://github.com/dotnet/runtime/blob/v8.0.1/src/libraries/System.Memory/tests/ParsersAndFormatters/Parser/TestData.Parser.TimeSpan.cs#L181-L206

You don't need to test the cases of leading whitespaces, negative values, and illegal cases. Such cases should never happen.

File: lib/auxiliary/time_helper.rb (hunk 2 above)
Comment (kostyanf14) [reply lib/auxiliary/time_helper.rb:8]: @akihikodaki I am ok with adding tests, just a question. Do we need to think about these formats:

- dd
- hh:mm
- dd.hh:mm
- hh:mm:ss.fffffff
- dd.hh:mm:ss.fffffff

All formats looks ok for MS, but I don't think HLK will provide something in its. Or at least ignore `second fraction`.

Comment (akihikodaki) [reply lib/auxiliary/time_helper.rb:8]: We will never have the following according to the documentation https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-timespan-format-strings and source code https://github.com/dotnet/runtime/blob/v8.0.1/src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanFormat.cs
- dd
- hh:mm
- dd.hh:mm

S second fraction is still possible. I think it's easy to convert to use a floating point number or just to ignore though (`to_i` will ignore the fraction).

File: spec/lib/auxiliary/test_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+require_relative '../../../lib/auxiliary/time_helper.rb'
```
Comment (kostyanf14) [inline spec/lib/auxiliary/test_helper_spec.rb:1]: `s/test_helper_spec.rb/time_helper_spec.rb/`

File: spec/lib/auxiliary/test_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+RSpec.describe AutoHCK::Helper do
+  include AutoHCK::Helper
+
+  describe '#time_to_seconds' do
+    
+    time_strings = {
+      "1:2" => 3720, #hh:mm
```
Comment (kostyanf14) [inline spec/lib/auxiliary/test_helper_spec.rb:9]: I don't prefer to have magic numbers without comments. 

You wrote `"1.2:3:4" => 86400 + 7380 + 4`, then I preferred `"1.2:3:4" => 86400 + 2 * 60 * 60 + 3 * 60 + 4` 
Other variants add comments before `time_strings` that the number is a second count and wrote `93784` instead of  `86400 + 7380 + 4`

Currently, it looks strange why we sometimes calculate something but sometimes have a constant.

File: spec/lib/auxiliary/test_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      "1:2:3" => 3723, #hh:mm:ss
+      "1.2:3" => 86400 + 7380, #dd.hh:mm
+      "1.2:3:4" => 86400 + 7380 + 4, #dd.hh:mm:ss
+      "1:2:3.22" => 3723.22, #hh:mm:ss.ff
+      "1.2:3:4.22" => 86400 + 7380 + 4.22, #dd.hh:mm:ss.ff
+    }
+
+    time_strings.each do |time, expected|
```
Comment (kostyanf14) [inline spec/lib/auxiliary/test_helper_spec.rb:17]: what is `expected`?  better `|time_string, seconds|`

File: lib/auxiliary/time_helper.rb [hunk 8]
```diff
@@ -4,9 +4,18 @@
 module AutoHCK
   # Helper module
   module Helper
-    def time_to_seconds(time)
-      time.split(':').reverse.map.with_index { |a, i| a.to_i * (60**i) }
-          .reduce(:+)
+    def time_to_seconds(time_string)
+      match = time_string.match(/^(?:(\d+)\.)?(?:(\d+):)?(\d+)(?::(\d+))?(?:\.(\d+))?$/)
```
Comment (akihikodaki) [inline lib/auxiliary/time_helper.rb:8]: To match entire second including fraction:

```suggestion
      match = time_string.match(/^(?:(\d+)\.)?(?:(\d+):)?(\d+)(?::(\d+(?:\.\d+)?))?$/)
```

File: spec/lib/auxiliary/test_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require_relative '../../../lib/auxiliary/time_helper'
+
+RSpec.describe AutoHCK::Helper do
+  include AutoHCK::Helper
+
+  describe '#time_to_seconds' do
+    time_strings = {
```
Comment (akihikodaki) [inline spec/lib/auxiliary/test_helper_spec.rb:9]: It is a bit awkward to name `time_strings` as it doesn't only have strings but also numbers. They are test cases so we can call `cases` or something like that.

File: spec/lib/auxiliary/test_helper_spec.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require_relative '../../../lib/auxiliary/time_helper'
+
+RSpec.describe AutoHCK::Helper do
```
Comment (akihikodaki) [inline spec/lib/auxiliary/test_helper_spec.rb:5]:
```suggestion
describe AutoHCK::Helper do
```

For consistency with other files. Not sure what this difference implies though.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: lib/auxiliary/time_helper.rb (hunk 3 above)
Comment (Jedoku) [reply lib/auxiliary/time_helper.rb]: i believe that code was more explicit and easier to understand at a glance

**Comment by kostyanf14** [comment]
@Jedoku LGTM

Please combine all your changes into one or two commits. For each review iteration, you should fix the original commits.

**Comment by kostyanf14** [comment]
@Jedoku You missed `Signed-off-by` in commit

**Review by kostyanf14 — APPROVED** [review]
